source/Missions/Services/Agents/Messages/AgentMovementDelta.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Missions.Services.Agents.Packets;
+using System;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Messages
+{
+    /// <summary>
+    /// Represents the delta of an agent's movement between the time it was created and the time it was last updated.
+    /// </summary>
+    internal sealed class AgentMovementDelta
+    {
+        private Vec3 _lookDirectionDelta = Vec3.Zero;
+        private Vec2 _inputVectorDelta = Vec2.Zero;
+
+        private AgentActionData _actionDataDelta;
+        private AgentMountData _mountDataDelta;
+
+        private Vec3 _currentPosition;
+
+        /// <summary>
+        /// Guid of the associated <see cref="Agent"/>.
+        /// </summary>
+        public Guid Guid { get; }
+
+        /// <summary>
+        /// The <see cref="Agent"/> whose movement is about to be updated.
+        /// </summary>
+        public Agent Agent { get; }
```

Comment:
Since this will be sent over the network Agent isn't needed.
```suggestion

```

Reply:
I will remove Agent and in it's stead will add Vec2 _agentMovementDirection and AgentEquipmentData as they are required for `AgentData`